cmd/callVariants/callVariants.go

```diff
@@ -18,133 +19,130 @@ import (
 
 func usage() {
 	fmt.Print(
-		"callVariants - Calls variants from input sam or giraf based on a linear or graph reference.\n" +
+		"callVariants - A tool to find variation between multiple alignment files.\n\n" +
 			"Usage:\n" +
-			" callVariants [options] \n" +
-			"\t-i experimental.sam \n" +
-			"\t-n normal.sam \n" +
-			"\t-lr reference.fasta \n" +
-			"\t-o output.vcf \n\n" +
-			"options:\n")
+			"  callVariants [options] -i file1.bam -i file2.bam -n normal.bam -r reference.fasta\n\n" +
+			"Options:\n\n")
 	flag.PrintDefaults()
 }
 
-func callVariants(linearRef string, graphRef string, expSamples string, normSamples string, outFile string, afThreshold float64, sigThreshold float64, minMapQ int64, memBufferSize int, minCov int) {
-	var ref interface{}
-	output := fileio.MustCreate(outFile)
-	defer output.Close()
+func callVariants(experimentalFiles, normalFiles []string, refFile, outFile string, maxP, minAf float64, minCoverage int) {
+	out := fileio.EasyCreate(outFile)
+	outHeader := makeOutputHeader(append(experimentalFiles, normalFiles...))
+	vcf.NewWriteHeader(out, outHeader)
+	ref := fasta.NewSeeker(refFile, "")
 
-	if linearRef != "" {
-		ref = fasta.Read(linearRef)
-	} else if graphRef != "" {
-		ref = genomeGraph.Read(graphRef)
+	expHeaders, expPiles := startPileup(experimentalFiles)
+	normHeaders, normPiles := startPileup(normalFiles)
+
+	isExperimental := make([]bool, len(experimentalFiles)+len(normalFiles))
+	for i := 0; i < len(experimentalFiles); i++ {
+		isExperimental[i] = true
 	}
-	alleleStream, normalIDs := startAlleleStreams(ref, expSamples, normSamples, minMapQ, memBufferSize)
-	answer := alleles.FindNewVariation(alleleStream, normalIDs, afThreshold, sigThreshold, minCov)
 
-	lastProgressReport := time.Now()
+	err := checkHeadersMatch(append(expHeaders, normHeaders...))
```

Comment:
What does this "..." do at the end here?

Reply:
the append function takes a slice (expHeaders) and adds single elements to the slice. But when you want to append one slice to another it doesn't work. You can get around that by expanding your slice to individual values using `...`. This line is equivalent to 
`append(expHeaders, normHeaders[0], normHeaders[1], normHeaders[2], normHeaders[n])`